pkg/config/config.go

```diff
@@ -37,38 +37,63 @@ var (
 	validRoles = []string{"controlplane", "node"}
 )
 
-type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+// Top level config
+type Config struct {
+	Core Core `json:core`
 }
 
-type DebugConfig struct {
-	Pprof bool `json:"pprof"`
+// Core config
+type Core struct {
+	Network   Network   `json:network`
+	DNS       DNS       `json:dns`
+	Debugging Debugging `json:debugging`
 }
 
-type MicroshiftConfig struct {
-	ConfigFile string `json:"configFile"`
-	DataDir    string `json:"dataDir"`
-
-	AuditLogDir string `json:"auditLogDir"`
-	LogVLevel   int    `json:"logVLevel"`
-
-	Roles []string `json:"roles"`
+type Network struct {
+	// IP address pool to use for pod IPs.
+	// This field is immutable after installation.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	// This field is immutable after installation.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// The port range allowed for Services of type NodePort.
+	// If not specified, the default of 30000-32767 will be used.
+	// Such Services without a NodePort specified will have one
+	// automatically allocated from this range.
+	// This parameter can be updated after the cluster is
+	// installed.
+	// +kubebuilder:validation:Pattern=`^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])-([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`
+	ServiceNodePortRange string `json:"serviceNodePortRange,omitempty"`
+
+	MicroshiftNetwork MicroshiftNetwork `json:",inline"`
```

Comment:
In the other PR we talked about moving MTU to a separate config file. I think that would be more consistent with our plans to allow different CNI plugins to be swapped in by the community. @zshi-redhat are you OK with that?

Reply:
Based on the suggestion here, this [PR 956](https://github.com/openshift/microshift/pull/956/) should be able to contain the MTU information.

Reply:
I think that adding support for the separate configuration file in another PR makes sense. Whether that's #956 or something else is up to @zshi-redhat

Reply:
@dinhxuanvu @dhellmann I will update #965 to include mtu in ovnk CNI config file.

Reply:
I updated the CNI PR (#956) to include the mtu config change.